File: proselint/checks/misc/chatspeak.py
"""
Chatspeak.

---
layout:     post
source:     ???
source_url: ???
title:      textese
date:       2014-06-10
categories: writing
---

Chatspeak.

"""

from __future__ import annotations

from proselint.checks import CheckSpec, Existence

examples_pass = [
    "Smoke phrase with nothing flagged.",
]

examples_fail = [
    "BRB getting coffee.",
]

check = CheckSpec(
    Existence([
        "2day",
        "4U",
        "AFAIK",
        "AFK",
        "AFK",
        "ASAP",
        "B4",
        "brb",
        "btw",
        "cya",
        "GR8",
        "lol",
        "LOL",
        "LUV",
        "OMG",
        "rofl",
        "roftl",
        "sum1",
        "SWAK",
        "THNX",
        "THX",
        "TTYL",
        "XOXO",
    ]),
    "misc.chatspeak",
    "'{}' is chatspeak. Write it out.",
)

__register__ = (check,)

File: proselint-rs/crates/proselint-checks/src/industrial_language/chatspeak.rs
use proselint_registry::checks::{Check, types::*, Padding};

pub const EXAMPLES_PASS: &[&str] = &["Smoke phrase with nothing flagged."];
pub const EXAMPLES_FAIL: &[&str] = &["BRB getting coffee."];

const CHECK: Check = Check {
	check_type: &Existence {
		items: &[
			"2day",
			"4U",
			"AFAIK",
			"AFK",
			"AFK",
			"ASAP",
			"B4",
			"brb",
			"btw",
			"cya",
			"GR8",
			"lol",
			"LUV",
			"OMG",
			"rofl",
			"rotfl",
			"sum1",
			"SWAK",
			"THNX",
			"THX",
			"TTYL",
			"XOXO",
		],
		padding: Padding::WordsInText,
		exceptions: &[],
	},
	path: "industrial_language.chatspeak",
	msg: "'{}' is chatspeak. Write it out.",
	..Check::default()
};

pub const REGISTER: &[Check] = &[CHECK];
